src/solvers/moonorsun.ts

```diff
@@ -31,6 +31,13 @@ const solve = async ({ Implies, Sum }: Context, puzzle: Puzzle, cs: Constraints,
                 .eq(whichSymbol.get(regions.get(p)).eq(symbols.findIndex(s => s.eq(symbol))))
         );
     }
+    regions.forEach(region => {
+        symbols.forEach((symbol, index) => {
+            if (region.filter(p => puzzle.symbols.get(p)?.eq(symbol)).length == 0) {
+                cs.add(whichSymbol.get(region).neq(index))
+            }
+        });
+    })
```

Comment:
Good catch.

I would find this logic more intuitive (comment moved from above):
```
    // All regions must have at least one moon or sun that is used by the loop
    for (const region of regions) {
        cs.add(Or(...region.filter(p => puzzle.symbols.has(p)).map(p => grid.get(p).neq(0))));
    }
```